Detections/SigninLogs/AuthenticationAttemptfromNewCountry.yaml

```diff
@@ -22,38 +22,59 @@ relevantTechniques:
-    | extend Result = strcat(tostring(ResultType), " - ", ResultDescription)
-    | summarize FirstSeen=min(TimeGenerated), LastSeen=max(TimeGenerated), make_set(Result), make_set(IPAddress), make_set(UserAgent), make_set(Place), make_set(DeviceId) by UserPrincipalName, Location, Category
+  // Combine two sign-in logs tables, AADNonInteractiveUserSignInLogs and SigninLogs
+  let CombinedSignInLogs = union isfuzzy=True AADNonInteractiveUserSignInLogs, SigninLogs;
+  // Fetch Azure IP address ranges data from a JSON file hosted on GitHub
+  let AzureRanges = externaldata(changeNumber: string, cloud: string, values: dynamic)
+  ["https://raw.githubusercontent.com/microsoft/mstic/master/PublicFeeds/MSFTIPRanges/ServiceTags_Public.json"] with(format='multijson')
+  | mv-expand values
+  | extend Name = values.name, AddressPrefixes = tostring(values.properties.addressPrefixes);
+  // Identify known locations to be excluded from analysis
+  let ExcludedKnownLocations = CombinedSignInLogs
+  | where TimeGenerated between (ago(14d)..ago(1d))
+  | where ResultType == 0
+  | summarize by Location;
+  // Find sign-in locations matching specific criteria
+  let MatchedLocations = materialize(CombinedSignInLogs
+  | where TimeGenerated > ago(1d)
+  | where ResultType !in (50126, 50053, 50074, 70044)
+  | where Location !in (ExcludedKnownLocations));
+  // Match IP addresses of matched locations with Azure IP address ranges
+  let MatchedIPs = MatchedLocations
+  | evaluate ipv4_lookup(AzureRanges, IPAddress, AddressPrefixes)
+  | project IPAddress;
+  // Exclude IP addresses that are already matched with Azure IP address ranges
+  let MaxSetSize = 5; // Set the maximum size limit for make_set
+  let ExcludedIPs = MatchedLocations
+  | where not (IPAddress in (MatchedIPs))
+  | where isnotempty(Location)
+  | extend LocationDetails_dynamic = column_ifexists("LocationDetails_dynamic", "")
+  | extend DeviceDetail_dynamic = column_ifexists("DeviceDetail_dynamic", "")
+  | extend LocationDetails = iif(isnotempty(LocationDetails_dynamic), LocationDetails_dynamic, parse_json(LocationDetails_string))
+  | extend DeviceDetail = iif(isnotempty(DeviceDetail_dynamic), DeviceDetail_dynamic, parse_json(DeviceDetail_string))
+  | extend City = tostring(LocationDetails.city)
+  | extend State = tostring(LocationDetails.state)
+  | extend Place = strcat(City, " - ", State)
+  | extend DeviceId = tostring(DeviceDetail.deviceId)
+  | extend Result = strcat(tostring(ResultType), " - ", ResultDescription)
+  | summarize FirstSeen=min(TimeGenerated), LastSeen=max(TimeGenerated), 
+    make_set(Result, MaxSetSize), make_set(IPAddress, MaxSetSize), 
+    make_set(UserAgent, MaxSetSize), make_set(Place, MaxSetSize), 
+    make_set(DeviceId, MaxSetSize) by UserPrincipalName, Location, Category;
+    ExcludedIPs
 entityMappings:
   - entityType: Account
     fieldMappings:
-      - identifier: FullName
+      - identifier: Name
         columnName: UserPrincipalName
```

Comment:
You will have to split by '@' and then get Name and UPNSuffix.

Reply:
did the suggested change about the name and upn suffix, please checout, 

@devikamehra